### ISSUE TITLE
Visual Studio 15.6.3 assets file doesn't have a target

### ISSUE THREAD

**Issue by ijsgaus** [dotnet/sdk#2068]
When build depended multitargeted (Net47, NetStandrd2.0) projects in solution in Visual Sudio, 
error:
Assets file 'BadCompile\ClassLibrary1\obj\project.assets.json' doesn't have a target for '.NETStandard,Version=v2.0'. Ensure that restore has run and that you have included 'netstandard2.0' in the TargetFrameworks for your project.	ClassLibrary1	C:\Program Files\dotnet\sdk\2.1.102\Sdks\Microsoft.NET.Sdk\build\Microsoft.PackageDependencyResolution.targets	167	

If switch build to single process mode - build success. Success build in command line tool
Sample solution: https://github.com/ijsgaus/BadSolution

**Comment by hemiaoio** [comment]
I also have this problem, my target frameworks is Net461 and NetStandrd2.0 . Delete the project.assets.json also does not help.

**Comment by risfeng** [comment]
I'm the same problem！Can you tell me how to solve it?

**Comment by hemiaoio** [comment]
Well, One of the friends said that it was caused by ReSharper. After it was disabled, it was normal. Have you also installed ReSharper?

**Comment by hemiaoio** [comment]
I was uninstalled ReSharper，it works。
And you can change then build tools from ReSharper options.
![qq 20180321132938](https://user-images.githubusercontent.com/5441896/37695667-ef33ca7c-2d0b-11e8-97d6-4139841e5a2f.png)
![qq 20180321132943](https://user-images.githubusercontent.com/5441896/37695669-f1b648a6-2d0b-11e8-8293-b9412fc29663.png)

**Comment by hemiaoio** [comment]
Another solution is to uninstall SDK 2.1.102 from your installed program。

**Comment by risfeng** [comment]
I'm the same problem.
Probably caused by the installation of the ReSharper plug-in, my solution is to use ReSharper build, set ReSharper build: **ReSharper-> Tools -> Build & Run.** This solves my problem, may wish you also try.

**Comment by ahawes-clarity** [comment]
Possibly related issue.

Assets file '...\obj\project.assets.json' doesn't have a target for '.NETFramework,Version=v4.6.2/win7-x64'. Ensure that restore has run and that you have included 'net462' in the TargetFrameworks for your project. You may also need to include 'win7-x64' in your project's RuntimeIdentifiers.

I only get this error when trying to Publish to Azure. I can build the Solution manually without issue.

I have confirmed I have included 'net462' in the TargetFrameworks. Actually, I am using the singular TargetFramework and 4.6.2 is the only target. I added <RuntimeIdentifier>win7-x64</RuntimeIdentifier> which was not needed previously and I have confirmed that the project.assets.json file does include a ".NETFramework,Version=v4.6.2/win7-x64" section.

This is blocking me from publishing some needed updates to an application.

**Comment by hemiaoio** [comment]
Well, eventually I uninstalled ReSharper to solve this problem. My colleague is solving this problem after uninstalling SDK 2.1.102.

**Comment by davkean** [comment]
We've figured out the issue internally; an assembly resolver in the R# extension is resolving one of our assemblies to one of their assemblies. This only occurs when .NET Core SDK 2.1.102 is installed, which comes with 15.6.3. 

We're working on a patch will be released in the next couple of days that will avoid R#'s assembly resolver to be called, and we're going to be working with R# to resolve the underlying issue.

**Comment by asvishnyakov** [comment]
May confirm what the problem with "\obj\project.assets.json' doesn't have a target for 'netcoreapp2.0'" caused by 2.1.102 version of SDK. Disabling ReSharper doesn't help, Visual Studio build used.

**Comment by davkean** [comment]
When you disabled R# did you restart VS?

**Comment by mgrishkov** [comment]
Faced with the same problem after updating VS 2017 up to 15.6.3, Uninstalling SDK 2.1.102 fix it.
Unfortunately, It isn't the first time when VS update is cause of unpredictable crash... sad...

**Comment by davkean** [comment]
@mgrishkov Glad you resolved the issue, disabling R# or uninstalling SDK 2.1.102 will temporarily resolve the issue. We have a fix coming tomorrow that prevents our assembly from being redirected by the R# assembly resolver and we'll work with JetBrains to have a more permanent fix.

**Comment by dotnetshadow** [comment]
@davkean I uninstalled SDK 2.1.102, will tomorrows fix install a new version of this through 15.6.4?

**Comment by davkean** [comment]
I'm unsure if it will be "branded" 15.6.4 but a new version of the SDK will be installed.

**Comment by mvelosop** [comment]
Don't you use R# at MS? ;-)

**Comment by mgrishkov** [comment]
@davkean, I don't use R#, I catch it with netstandard1.6 lib. project and netcoreapp1.6 mstest project.

**Comment by amccool** [comment]
I uninstalled SDK 2.1.102, and repaired 2.1.101.      ---- However my system didnt delete the ` C:\Program Files\dotnet\sdk\2.1.102` directory fully (it left the bin and obj dirs)

once I did rmdir ` C:\Program Files\dotnet\sdk\2.1.102`   then my multi-targeted builds started working again

**Comment by jeoffman** [comment]
@mgrishkov and @amccool had a good tip, close VS (wait for it to fully shutdown) and then uninstall the SDK 2.1.102 - I can build again. Thanks guys!

**Comment by davkean** [comment]
@mvelosop We discovered the issue internally before we shipped, and JetBrains made a change in response to it, but we were unaware of their assembly resolver - so our recommendations were incomplete.

**Comment by softwarebybrent** [comment]
So what's the overall state of this? I saw [here](https://youtrack.jetbrains.com/issue/RSRP-468758#u=1521542921889) that they are working on getting a fix out for it, but it appears to have quiet down since this morning.  Are we expecting a 15.6.4 of VS at some point in conjunction with R# release or just waiting on R# to fix what they need to?

**Comment by davkean** [comment]
We are making a temporary fix that will unblock without a JetBrains. JetBrains is making a long term fix that will prevent this situation from happening again.

Our is still in testing, and should be available in the next day or two.

**Comment by johnkors** [comment]
Just to add to this. Suspending ReSharper, rebooting VS, rebooting the machine.. did not help. 

![image](https://user-images.githubusercontent.com/206726/37752698-9b07ab42-2d99-11e8-9dbc-2cb51f149e18.png)

Repro sln: https://github.com/johnkors/vs2017-1563

**Comment by darkonejr** [comment]
@johnkors You need to fully disable R#.

Fwiw, setting the number of parallel builds to 1 seems to work for me as a workaround as well.

**Comment by johnkors** [comment]
Right, thanks. That was the trick. Thanks @darkonejr . Updated my comment.

**Comment by livarcocc** [comment]
This has now been fixed in the SDK 2.1.103 and VS 15.6.4.

**Comment by AndreyAkinshin** [comment]
The problem is also fixed in ReSharper 2017.3.5, everything works with all combinations of Visual Studio and .NET Core SDK.
You can find more technical details in the official blog post: https://blog.jetbrains.com/dotnet/2018/03/23/build-failures-visual-studio-15-6-3-resharper-ultimate-2017-3-5-rescue/

**Comment by SimonCropp** [comment]
Thanks everyone

**Comment by mwpowellhtx** [comment]
To be clear, I do have R# installed, but I am encountering the error invoking MSBuild programmatically, ***NOT*** from any IDE session. So, as far as I know, R# shouldn't be in the picture?

**Comment by mwpowellhtx** [comment]
@livarcocc Sorry, this is not resolved. As far as I know, the latest similar ``sdk`` I have is ``2.1.201``, running VS ``15.7.4``, and when I invoke MSBuild programmatically I am getting the error:

```
Assets file '...\obj\project.assets.json' doesn't have a target for '.NETCoreApp,Version=v2.0'. Ensure that restore has run and that you have included 'netcoreapp2.0' in the TargetFrameworks for your project.
```

I can assure you that it not only has the file, but I even invoked the ``nuget.exe restore`` process preceding engagement with ``MSBuild``.

**Comment by mwpowellhtx** [comment]
So... I wanted to follow up. I ended up pulling my ``C:\Program Files\dotnet\sdk`` folders aside, both ``2.1.200`` and ``2.1.201``, and the programmatic invocation succeeded. From there I should be able to verify before and after states that I expect with my custom MSBuild tasks.

**Comment by tomgruszowski** [comment]
@mwpowellhtx this may not be an option for you but maybe upgrade to 2.1? It's quite a painless update and 2.0 is not the LTS branch, 2.1 is. FYI 2.1.1 SDK is just around the corner, available [https://github.com/aspnet/home/wiki/.NET-Core-2.1.1-Early-Access-Downloads](https://github.com/aspnet/home/wiki/.NET-Core-2.1.1-Early-Access-Downloads)

**Comment by SharePointSean** [comment]
It's broken again.

![image](https://user-images.githubusercontent.com/881994/47316689-b7234780-d615-11e8-929e-4eab33b751b5.png)

**Comment by livarcocc** [comment]
@SharePointSean do not assume that this is the same issue as above. Can you please file a new, separate issue for your problem with details on how to repro it so that we can investigate?

**Comment by tet-web-dev** [comment]
I don't have R# installed but I received this when I go to publish the project after changing the target framework from  2.0 to 2.1.
![image](https://user-images.githubusercontent.com/6007105/48589710-bc20a180-e90a-11e8-8a0d-2fa4d05d7e66.png)
